src/controller/java/src/chip/devicecontroller/model/ClusterState.java

```diff
@@ -51,6 +55,33 @@ public Optional<Long> getDataVersion() {
     return dataVersion;
   }
 
+  /**
+   * Convenience utility for getting all attributes in Json string format.
+   *
+   * @return all attributes in Json string format., or empty string if not found.
+   */
+  public String getAttributesJson() {
+    JSONObject combinedObject = new JSONObject();
+    attributes.forEach(
+        (attributeId, attributeState) -> {
+          JSONObject attributeJsonObj = attributeState.getJson();
+          if (attributeJsonObj != null) {
+            Iterator<String> keys = attributeJsonObj.keys();
+            while (keys.hasNext()) {
```

Comment:
can we just `for (key in attributeJsonObject.keys())`  ?

Reply:
Done, thanks